### PR TITLE
Add force_page_reload option to ObjectFormWidget

### PR DESCRIPTION
**Pull request by MouseEatsCat**
This option will force the form widget to reload the page after update (when set to: true).
Example:
```JSON
"admin.edit": {
    "widgets": {
        "form": {
            "type": "charcoal/admin/widget/object-form",
            "form_ident": "default",
            "obj_type": "charcoal/admin/object/object-name",
            "group_display_mode": "tab",
            "force_page_reload": true
        }
    }
}
```

### CODE REVIEW

File: src/Charcoal/Admin/Widget/ObjectFormWidget.php [hunk 1]
```diff
@@ -344,7 +370,8 @@ public function widgetDataForJs()
             'tab'                => $this->isTabbable(),
             'group_display_mode' => $this->groupDisplayMode(),
             'group_conditions'   => $this->groupsConditionalLogic(),
-            'allow_reload'       => $this->allowReload()
+            'allow_reload'       => $this->allowReload(),
+            'force_refresh'       => $this->forceRefresh()
```
Comment (mcaskill) [inline src/Charcoal/Admin/Widget/ObjectFormWidget.php]: Fix alignment of `key => value` assignment operators and add trailing comma for multi-line array declarations.

```suggestion
            'allow_reload'       => $this->allowReload(),
            'force_refresh'      => $this->forceRefresh(),
```

File: assets/src/scripts/charcoal/admin/widget/form.js [hunk 2]
```diff
@@ -50,6 +50,7 @@ Charcoal.Admin.Widget_Form.prototype.set_properties = function (opts) {
     this.group_conditions = opts.data.group_conditions;
     this.$form            = $(this.form_selector);
     this.allow_reload     = opts.data.allow_reload;
+    this.force_refresh     = opts.data.force_refresh;
```
Comment (mcaskill) [inline assets/src/scripts/charcoal/admin/widget/form.js]: I think we should rename this property to `force_page_refresh` to distinguish from `allow_reload` which should have been named `force_widget_reload` (to be resolved in a future changeset).

```suggestion
    this.force_page_refresh = opts.data.force_page_refresh;
```

Don't forget to fix alignment of `=` assignment operators.

File: assets/src/scripts/charcoal/admin/widget/form.js [hunk 3]
```diff
@@ -435,6 +436,10 @@ Charcoal.Admin.Widget_Form.prototype.request_success = function ($form, $trigger
                 '&obj_id=' + response.obj_id;
         }
     } else {
+        if (this.force_refresh) {
+            window.location.href = window.location.href;
+        }
+
```
Comment (mcaskill) [inline assets/src/scripts/charcoal/admin/widget/form.js:442]: If the response has feedback (`response.feedbacks`), the force refresh should occur after a confirmation alert, similar to `response.next_url` above.

```suggestion
        if (this.force_refresh) {
			if (response.feedbacks) {
				// @todo "dynamise" the label
				Charcoal.Admin.feedback().add_action({
					label:    commonL10n.continue,
					callback: function () {
						window.location.reload();
					}
				});
			} else {
                window.location.reload();
            }
        }

```

Also, `window.location.reload()` seems like the better option for reloading the page.

File: assets/src/scripts/charcoal/admin/widget/form.js [hunk 4]
```diff
@@ -50,6 +50,7 @@ Charcoal.Admin.Widget_Form.prototype.set_properties = function (opts) {
     this.group_conditions = opts.data.group_conditions;
     this.$form            = $(this.form_selector);
     this.allow_reload     = opts.data.allow_reload;
+    this.force_refresh    = opts.data.force_refresh;
```
Comment (mcaskill) [inline assets/src/scripts/charcoal/admin/widget/form.js:53]: I think this property should be called `force_page_reload` to clarify its purpose, match the [`Location.reload()` API](https://developer.mozilla.org/en-US/docs/Web/API/Location/reload), and distinguish itself from `allow_reload`, which should have been called `force_widget_reload`.

```suggestion
    this.force_page_reload = opts.data.force_page_reload;
```

**Review by mcaskill — CHANGES_REQUESTED** [review]

File: src/Charcoal/Admin/Widget/ObjectFormWidget.php [hunk 5]
```diff
@@ -344,7 +370,8 @@ public function widgetDataForJs()
             'tab'                => $this->isTabbable(),
             'group_display_mode' => $this->groupDisplayMode(),
             'group_conditions'   => $this->groupsConditionalLogic(),
-            'allow_reload'       => $this->allowReload()
+            'allow_reload'       => $this->allowReload(),
+            'force_refresh'      => $this->forceRefresh()
```
Comment (mcaskill) [inline src/Charcoal/Admin/Widget/ObjectFormWidget.php:374]: Add trailing comma for multi-line array declarations.

```suggestion
            'force_refresh'      => $this->forceRefresh(),
```

**Comment by MouseEatsCat** [comment]
@mcaskill My logic is that the force refresh should behave similarly to the `this.is_new_object` when it redirects you to the edit page when you hit save (on a "create" form).

**Comment by mcaskill** [comment]
> @mcaskill My logic is that the force refresh should behave similarly to the `this.is_new_object` when it redirects you to the edit page when you hit save (on a "create" form).

I completely agree.

**Review by mcaskill — APPROVED** [review]

**Pull request merged**